### PR TITLE
Do not connect to peers with impossibly high heads

### LINKED ISSUE #1986: don't attempt syncing to chains whose height is obviously wrong
Since we reject blocks from the future, and have a 1:1 mapping between block height a timestamp, we should reject blocks whose height is too high to link to our genesis block.

As i type this, my interopnet node is attempting to sync to either the testnet or an older interopnet.

### PR DESCRIPTION
**Pull request by arajasek**
- Arbitrarily put a drift of 5, this should probably be some constant
- The `Syncer` is a somewhat odd place to put the function, open to hearing suggestions

Closes #1986

### CODE REVIEW

File: node/hello/hello.go [hunk 1]
```diff
@@ -79,6 +79,17 @@ func (hs *Service) HandleStream(s inet.Stream) {
 		s.Conn().Close()
 		return
 	}
+
+	maxHeight, err := hs.syncer.MaxHeightAtCurrentTime()
+	if err != nil {
+		log.Warnf("failed to calculate max possible height: %w", err)
+		s.Conn().Close()
+		return
+	} else if hmsg.HeaviestTipSetHeight > maxHeight+5 {
```
Comment (schomatis) [inline node/hello/hello.go:88]: Please extract and document this constant.

File: node/hello/hello.go [hunk 2]
```diff
@@ -79,6 +79,17 @@ func (hs *Service) HandleStream(s inet.Stream) {
 		s.Conn().Close()
 		return
 	}
+
+	maxHeight, err := hs.syncer.MaxHeightAtCurrentTime()
+	if err != nil {
+		log.Warnf("failed to calculate max possible height: %w", err)
```
Comment (whyrusleeping) [inline node/hello/hello.go:85]: closing a connection because of a local error seems fishy, at the very least this should be an Error log.

**Comment by whyrusleeping** [comment]
As @schomatis says, let's also add this check to either the gossipsub validator, and/or the ValidateBlock function in the syncer.

File: node/hello/hello.go (hunk 2 above)
Comment (arajasek) [reply node/hello/hello.go:85]: Yeah, I'll definitely bump it to an Error...we could also just ignore this check if we error at `MaxHeight`, except we should only error there if we fail to load our genesis block, in which case big shrug

**Comment by whyrusleeping** [comment]
@arajasek update here?

**Comment by arajasek** [comment]
Moved the check into `InformNewHead`, that feels like the right place...It now lives entirely in Syncer, which also feels correct.


> 
> edit: actually this is already implemented in `ValidateBlock` so if we can unify both logic to make sure they represent the same validation (so we can later also apply it in `(*BlockValidator).Validate()`) i'm happy

@schomatis I don't think this is implemented in `ValidateBlock` (except indirectly, by comparing against parent tipset's timestamp)...But by that point you've already fetched and validated all the earlier blocks, which is what this is trying to avoid (those earlier blocks don't exist).

**Pull request merged**